src/darwin/frida-helper-backend-glue.m

```diff
@@ -1375,12 +1377,26 @@ static void frida_darwin_helper_backend_launch_using_sbs (NSString * identifier,
 
     timer = g_timer_new ();
 
-    while ([service pidForApplication:identifier] > 0 && g_timer_elapsed (timer, NULL) < kill_timeout)
+    while ((killed_pid = [service pidForApplication:identifier]) > 0 && g_timer_elapsed (timer, NULL) < kill_timeout)
```

Comment:
Does this mean we always get stuck here waiting three seconds if a prewarmed app is being killed? If so it might be worth doing the check earlier, so we can just `SIGKILL` it right away and not have to wait.

(That reminds me it's unfortunate that we block the current thread's `MainContext` here -- frida-server is not able to do anything else while we're stuck here, when this function is called from `DarwinHelperBackend._kill_process()` and `DarwinHelperBackend._kill_application()`.)

Reply:
for not blocking the thread i wonder if this could be made async, by registering a SEND_DEAD callback on the about-to-bi-killed pid's task? anyways i think that belongs to a separate PR in a future time, because it's quite complex and can have weird edge cases which require proper testing IMHO.

Reply:
The easiest would probably be implementing it in Vala as an async method, where it uses a `TimeoutSource` to repeatedly check. But yeah that part is better left for a separate future PR.

> Does this mean we always get stuck here waiting three seconds if a prewarmed app is being killed?

My concern with this is that we always wait three seconds if the app has been prewarmed. Could we do the `has_active_prewarm()` before and go straight to `SIGKILL` in that case, or is this a bad idea?

Reply:
Oh wait, sorry, that's exactly what you did :blush:

Reply:
yeah i initially implemented it differently because i thought the sysctl call would have been slow, instead is ridiculously fast 🤣